toptal/picasso
# feat: add missing icons and enforce icon sizes

[FX-4158]

### Description

According to BASE design (https://www.figma.com/file/TqaGgbpjGSUDf7qq153Isq/Product-Iconography?type=design&node-id=1-2969&mode=design&t=Sz5fcUd1rnrqmlWx-0), icons should always have `16px` and `24px` versions (https://toptal-core.slack.com/archives/C05CSR5RFBN/p1688574324432869?thread_ts=1687249523.324339&cid=C05CSR5RFBN).

This pull request adds missing icons (taken from BASE) and adds a pre-commit check that makes sure that icon is added.

⚠️ Please note that `Ach` and `CreditCard` will be added after they are added to BASE

### How to test

- Check the icon list at https://picasso.toptal.net/FX-NULL-add-missing-icons/?path=/story/components-icons--icons
- Try to push incomplete icon locally

<img width="612" alt="Screenshot 2023-07-05 at 20 04 03" src="https://github.com/toptal/picasso/assets/1390758/a91a80f6-cda9-4d3e-8eb8-36652d664072">


### Development checks

- [N/A] Annotate all `props` in component with documentation
- [N/A] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).
- [N/A] Covered with tests

**Breaking change**

- [N/A] codemod is created and showcased in the changeset
- [N/A] test alpha package of Picasso in StaffPortal

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>


[FX-4158]: https://toptal-core.atlassian.net/browse/FX-4158?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
